lib/src/rpc/dial.dart

```diff
@@ -44,6 +46,9 @@ class DialOptions {
   /// If enabled, other auth options have no affect. Eg. [authEntity], [credentials],
   /// [externalAuthAddress], [externalAuthToEntity]
   String? accessToken;
+
+  /// Whether the connection was made usuing MDNS
+  bool usingMdns = false;
```

Comment:
Can this be private? IMO, `DialOptions` should be used to allow users to set options, not for internal state-keeping.

Also, I think we should have an option called `useMdns` that allows users to turn on/off mdns.

Reply:
i agree this should be private, I initially didn't do that because goUtils doesn't have it private. But you're right it ought to be. +1 on adding the flag for enabling it, our other implementations also have that, I will add. thank you!